File: 1814. Count Nice Pairs in an Array/Solution.cpp
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    int countNicePairs(vector<int>& nums)
    {
        int res = 0, MOD = 1e9 + 7;
        unordered_map<int, int> cnt;
        for (const int& x : nums) {
            int rx = x - rev(x);
            res = (res + cnt[rx]) % MOD;
            ++cnt[rx];
        }
        return res;
    }

private:
    int rev(int x)
    {
        int res = 0;
        while (x) {
            res = res * 10 + x % 10;
            x /= 10;
        }
        return res;
    }
};

File: 1814. Count Nice Pairs in an Array/Solution.py
from collections import defaultdict
from typing import List


class Solution:
    def countNicePairs(self, nums: List[int]) -> int:
        res, MOD = 0, 10**9 + 7
        cnt = defaultdict(int)
        for e in nums:
            x = e - int(str(e)[::-1])
            res = (res + cnt[x]) % MOD
            cnt[x] += 1
        return res
